crates/provider-sdk/src/lib.rs

```diff
@@ -7,25 +7,35 @@ use async_trait::async_trait;
 use serde::{Deserialize, Serialize};
 use tracing::{error, info, warn};
 
+use crate::error::{InvocationError, InvocationResult};
+
 pub mod error;
 pub mod provider;
-pub mod provider_main;
-pub mod rpc_client;
-
-pub use provider::ProviderConnection;
-pub use provider_main::{load_host_data, run_provider, start_provider};
-pub use rpc_client::RpcClient;
-
-pub use wasmcloud_tracing;
 
 /// Re-export of types from [`wasmcloud_core`]
 pub use core::{
     HealthCheckRequest, HealthCheckResponse, InterfaceLinkDefinition, WasmCloudEntity, WitFunction,
     WitInterface, WitNamespace, WitPackage,
 };
+pub use provider::*;
```

Comment:
NIT: it's better to have our exports be explicit here so we can be very sure what we expose/don't expose to the SDK.